### DISCUSSION TITLE
Tiffsave with bands and pyramid layer in separated IFDs

### DISCUSSION THREAD

**Discussion by gil-c** [libvips/libvips#2868] [Q&A]
Hello,

I am trying to generate an ome.tiff file with a similar file structure than this sample image: https://www.synapse.org/#!Synapse:syn17778717

From what I can see from tiffinfo, the file is organized with multiple IFDs, all having Samples/Pixel: 1 (one band).

The first 44 IFDs are 44 bands in the original resolution, the 44 following IFDs are the same bands with half resolution, and so on. In total the file has 44 channels times 5 resolutions equals 220 IFDs. 

I cannot think of a way to reproduce such tiff structure with libvips using the combinations of tiffsave arguments like pyramid, sub-ifd, or page_height. Is there a way? 

Kind regards

**Comment by jcupitt** [comment]
Hello @gil-c,

You need to make a multipage TIFF, one band per page (so 44 in this case), then save as an ifd pyramid.

There's some sample code here:

https://forum.image.sc/t/writing-qupath-bio-formats-compatible-pyramidal-image-with-libvips/51223/6

You'll also need the XML in `image-description`.

You might find vipsdisp helpful:

https://flathub.org/apps/details/org.libvips.vipsdisp

It can display large images of this type quickly.

**Reply by gil-c** [reply]
Sounds great! I'll test this, thanks a lot!
